NuGet.Client build.sh is broken

`./build.sh` fails when trying to install .NET Core 1.1:
```
$ ./build.sh 
Installing dotnet CLI
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 34939  100 34939    0     0  85425      0 --:--:-- --:--:-- --:--:--  541k
dotnet_install: Warning: Unable to locate zlib. Probable prerequisite missing; install zlib.
dotnet-install: Downloading link: https://dotnetcli.azureedge.net/dotnet/Sdk/1.1.11/dotnet-sdk-1.1.11-linux-x64.tar.gz
curl: (22) The requested URL returned error: 404 
dotnet-install: Cannot download: https://dotnetcli.azureedge.net/dotnet/Sdk/1.1.11/dotnet-sdk-1.1.11-linux-x64.tar.gz
dotnet_install: Error: Could not find/download: `.NET Core SDK` with version = 1.1.11
dotnet_install: Error: Refer to: https://aka.ms/dotnet-os-lifecycle for information on .NET Core support
```

I am facing same issue while trying to build nuget client. A couple of questions before giving up on the build.sh.
1. What is the correct value for the channel (via "-c" param) within build.sh?
2. Do I need to give version (via "-v" param) within build.sh to fetch correct dotnet CLI?

I tried [2] and it did proceed further but only to fail later.
Private Changes:
```
neeraj@lp:~/other/NuGet.Client$ git diff build.sh
diff --git a/build.sh b/build.sh
index 565ef77c4..b36b82455 100755
--- a/build.sh
+++ b/build.sh
@@ -19,7 +19,7 @@ curl -o cli/dotnet-install.sh -L https://dot.net/v1/dotnet-install.sh
 chmod +x cli/dotnet-install.sh

 # v1 needed for some test and bootstrapping testing version
-cli/dotnet-install.sh -i cli -c 1.0
+cli/dotnet-install.sh -i cli -c 1.0 -v 5.0.101

 DOTNET="$(pwd)/cli/dotnet"

```

Failure logs:
```
neeraj@lp:~/other/NuGet.Client$ ./build.sh
Installing dotnet CLI
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 35418  100 35418    0     0  38708      0 --:--:-- --:--:-- --:--:-- 38708
dotnet-install: Note that the intended use of this script is for Continuous Integration (CI) scenarios, where:
dotnet-install: - The SDK needs to be installed without user interaction and without admin rights.
dotnet-install: - The SDK installation doesn't need to persist across multiple CI runs.
dotnet-install: To set up a development environment or to run apps, use installers rather than this script. Visit https://dotnet.microsoft.com/download to get the installer.

dotnet-install: Downloading link: https://dotnetcli.azureedge.net/dotnet/Sdk/5.0.101/dotnet-sdk-5.0.101-linux-x64.tar.gz
dotnet-install: Extracting zip from https://dotnetcli.azureedge.net/dotnet/Sdk/5.0.101/dotnet-sdk-5.0.101-linux-x64.tar.gz
dotnet-install: Adding to current process PATH: `/home/neeraj/other/NuGet.Client/cli`. Note: This change will be visible only when sourcing script.
dotnet-install: Note that the script does not resolve dependencies during installation.
dotnet-install: To check the list of dependencies, go to https://docs.microsoft.com/dotnet/core/install, select your operating system and check the "Dependencies" section.
dotnet-install: Installation finished successfully.
/home/neeraj/other/NuGet.Client/cli/dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting
  master 5.0.100-preview.7.20319.6
master 5.0.100-preview.7.20319.6
dotnet_install: Error: Unknown argument `5.0.100-preview.7.20319.6`
5.0.101
=================

...

EXEC : The path '/home/neeraj/other/NuGet.Client/packages/xunitxml.testlogger/2.0.0/build/_common' specified in the 'TestAdapterPath' is invalid. warning : The custom test adapter search path provided was not found, provide a valid path and try again. [/home/neeraj/other/NuGet.Client/build/build.proj]
/home/neeraj/other/NuGet.Client/build/build.proj(406,5): warning MSB3073: The command "/home/neeraj/other/NuGet.Client/cli/dotnet vstest /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/Microsoft.Build.NuGetSdkResolver.Tests/bin/Release/netcoreapp2.1/Microsoft.Build.NuGetSdkResolver.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/Microsoft.Build.NuGetSdkResolver.Tests/bin/Release/netcoreapp5.0/Microsoft.Build.NuGetSdkResolver.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Build.Tasks.Console.Test/bin/Release/netcoreapp2.1/NuGet.Build.Tasks.Console.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Build.Tasks.Console.Test/bin/Release/netcoreapp5.0/NuGet.Build.Tasks.Console.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Build.Tasks.Pack.Test/bin/Release/netcoreapp2.1/NuGet.Build.Tasks.Pack.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Build.Tasks.Pack.Test/bin/Release/netcoreapp5.0/NuGet.Build.Tasks.Pack.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Build.Tasks.Test/bin/Release/netcoreapp2.1/NuGet.Build.Tasks.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Build.Tasks.Test/bin/Release/netcoreapp5.0/NuGet.Build.Tasks.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.CommandLine.Xplat.Tests/bin/Release/netcoreapp2.1/NuGet.CommandLine.Xplat.Tests.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.CommandLine.Xplat.Tests/bin/Release/netcoreapp5.0/NuGet.CommandLine.Xplat.Tests.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Commands.Test/bin/Release/netcoreapp2.1/NuGet.Commands.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Commands.Test/bin/Release/netcoreapp5.0/NuGet.Commands.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Common.Test/bin/Release/netcoreapp2.1/NuGet.Common.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Common.Test/bin/Release/netcoreapp5.0/NuGet.Common.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Configuration.Test/bin/Release/netcoreapp2.1/NuGet.Configuration.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Configuration.Test/bin/Release/netcoreapp5.0/NuGet.Configuration.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.DependencyResolver.Core.Tests/bin/Release/netcoreapp2.1/NuGet.DependencyResolver.Core.Tests.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.DependencyResolver.Core.Tests/bin/Release/netcoreapp5.0/NuGet.DependencyResolver.Core.Tests.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Frameworks.Test/bin/Release/netcoreapp2.1/NuGet.Frameworks.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.LibraryModel.Tests/bin/Release/netcoreapp2.1/NuGet.LibraryModel.Tests.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Packaging.Test/bin/Release/netcoreapp2.1/NuGet.Packaging.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Packaging.Test/bin/Release/netcoreapp5.0/NuGet.Packaging.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.ProjectModel.Test/bin/Release/netcoreapp2.1/NuGet.ProjectModel.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.ProjectModel.Test/bin/Release/netcoreapp5.0/NuGet.ProjectModel.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Protocol.Tests/bin/Release/netcoreapp2.1/NuGet.Protocol.Tests.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Protocol.Tests/bin/Release/netcoreapp5.0/NuGet.Protocol.Tests.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Resolver.Test/bin/Release/netcoreapp2.1/NuGet.Resolver.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Resolver.Test/bin/Release/netcoreapp5.0/NuGet.Resolver.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Shared.Tests/bin/Release/netcoreapp2.1/NuGet.Shared.Tests.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Shared.Tests/bin/Release/netcoreapp5.0/NuGet.Shared.Tests.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Versioning.Test/bin/Release/netcoreapp2.1/NuGet.Versioning.Test.dll /home/neeraj/other/NuGet.Client/test/NuGet.Core.Tests/NuGet.Versioning.Test/bin/Release/netcoreapp5.0/NuGet.Versioning.Test.dll --TestAdapterPath:/home/neeraj/other/NuGet.Client/packages/xunitxml.testlogger/2.0.0/build/_common --logger:"xunit;LogFileName=CoreUnitTests-vsts.xml" --logger:"console;verbosity=detailed" --settings:/home/neeraj/other/NuGet.Client/build/xunit.runsettings" exited with code 1.
/home/neeraj/other/NuGet.Client/build/build.proj(406,5): warning MSB4181: The "Exec" task returned false but did not log an error.
/home/neeraj/other/NuGet.Client/build/build.proj(413,5): error : NETCore CoreUnitTests tests failed! Results: /home/neeraj/other/NuGet.Client/build/TestResults/CoreUnitTests-vsts.xml
Tests failed!!

```

--- a/.gitattributes
+++ b/.gitattributes
@@ -51,6 +51,9 @@ project.json text
 *.sql   text
 *.vb    text
 
+# Unix artifacts
+*.sh text eol=lf
+
 # Visual Studio artifacts
 *.csproj text eol=crlf
 *.dbproj text eol=crlf

--- a/build.sh
+++ b/build.sh
@@ -15,11 +15,21 @@ echo "Installing dotnet CLI"
 mkdir -p cli
 curl -o cli/dotnet-install.sh -L https://dot.net/v1/dotnet-install.sh
 
+if (( $? )); then
+	echo "Could not download 'dotnet-install.sh' script. Please check your network and try again!"
+	exit 1
+fi
+
 # Run install.sh for cli
 chmod +x cli/dotnet-install.sh
 
 # v1 needed for some test and bootstrapping testing version
 cli/dotnet-install.sh -i cli -c 1.0
+
+if (( $? )); then
+	echo "The .NET CLI Install failed!!"
+	exit 1
+fi
 
 DOTNET="$(pwd)/cli/dotnet"
 
@@ -32,8 +42,13 @@ DOTNET_BRANCH="$($DOTNET msbuild build/config.props /v:m /nologo /t:GetCliBranch
 echo $DOTNET_BRANCH
 cli/dotnet-install.sh -i cli -c $DOTNET_BRANCH
 
-# Display current version
-$DOTNET --version
+if (( $? )); then
+	echo "The .NET CLI Install failed!!"
+	exit 1
+fi
+
+# Display .NET CLI info
+$DOTNET --info
 
 echo "================="
 
@@ -54,6 +69,7 @@ fi
 # restore packages
 echo "$DOTNET msbuild build/build.proj /t:Restore /p:VisualStudioVersion=16.0 /p:Configuration=Release /p:BuildNumber=1 /p:ReleaseLabel=beta"
 $DOTNET msbuild build/build.proj /t:Restore /p:VisualStudioVersion=16.0 /p:Configuration=Release /p:BuildNumber=1 /p:ReleaseLabel=beta
+
 if [ $? -ne 0 ]; then
 	echo "Restore failed!!"
 	exit 1

--- a/build.sh
+++ b/build.sh
@@ -24,7 +24,7 @@ fi
 chmod +x cli/dotnet-install.sh
 
 # Get recommended version for bootstrapping testing version
-cli/dotnet-install.sh -i cli -c 1.0
+cli/dotnet-install.sh -i cli -c 2.2 -nopath
 
 if (( $? )); then
 	echo "The .NET CLI Install failed!!"
@@ -42,24 +42,34 @@ echo "dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting"
 IFS=$'\n'
 CMD_OUT_LINES=(`dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting`)
 # Take only last the line which has the version information and strip all the spaces
-CMD_LAST_LINE=${CMD_OUT_LINES[-1]//[[:space:]]}
+DOTNET_BRANCHES=${CMD_OUT_LINES[-1]//[[:space:]]}
 unset IFS
 
 IFS=$';'
-DOTNET_BRANCHES=(${CMD_LAST_LINE})
-unset IFS
+for DOTNET_BRANCH in ${DOTNET_BRANCHES[@]}
+do
+	echo $DOTNET_BRANCH
 
-IFS=$':'
-DOTNET_BRANCH=(${DOTNET_BRANCHES[0]})
-unset IFS
+	IFS=$':'
+	ChannelAndVersion=($DOTNET_BRANCH)
+	Channel=${ChannelAndVersion[0]}
+	if [ ${#ChannelAndVersion[@]} -eq 1 ]
+	then
+		Version="latest"
+	else
+		Version=${ChannelAndVersion[1]}
+	fi
+	unset IFS
 
-echo $DOTNET_BRANCH
-cli/dotnet-install.sh -i cli -c $DOTNET_BRANCH
+	echo "Channel is: $Channel"
+	echo "Version is: $Version"
+	cli/dotnet-install.sh -i cli -c $Channel -v $Version -nopath
 
-if (( $? )); then
-	echo "The .NET CLI Install failed!!"
-	exit 1
-fi
+	if (( $? )); then
+		echo "The .NET CLI Install for $DOTNET_BRANCH failed!!"
+		exit 1
+	fi
+done
 
 # Display .NET CLI info
 $DOTNET --info

--- a/build.sh
+++ b/build.sh
@@ -31,6 +31,9 @@ if (( $? )); then
 	exit 1
 fi
 
+# Disable .NET CLI Install Lookup
+DOTNET_MULTILEVEL_LOOKUP=0
+
 DOTNET="$(pwd)/cli/dotnet"
 
 # Let the dotnet cli expand and decompress first if it's a first-run

--- a/build.sh
+++ b/build.sh
@@ -23,7 +23,7 @@ fi
 # Run install.sh for cli
 chmod +x cli/dotnet-install.sh
 
-# v1 needed for some test and bootstrapping testing version
+# Get recommended version for bootstrapping testing version
 cli/dotnet-install.sh -i cli -c 1.0
 
 if (( $? )); then
@@ -33,11 +33,25 @@ fi
 
 DOTNET="$(pwd)/cli/dotnet"
 
-echo "$DOTNET msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting"
+# Let the dotnet cli expand and decompress first if it's a first-run
+$DOTNET --info
 
-# run it twice so dotnet cli can expand and decompress without affecting the result of the target
-$DOTNET msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting
-DOTNET_BRANCH="$($DOTNET msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting)"
+# Get CLI Branches for testing
+echo "dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting"
+
+IFS=$'\n'
+CMD_OUT_LINES=(`dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting`)
+# Take only last the line which has the version information and strip all the spaces
+CMD_LAST_LINE=${CMD_OUT_LINES[-1]//[[:space:]]}
+unset IFS
+
+IFS=$';'
+DOTNET_BRANCHES=(${CMD_LAST_LINE})
+unset IFS
+
+IFS=$':'
+DOTNET_BRANCH=(${DOTNET_BRANCHES[0]})
+unset IFS
 
 echo $DOTNET_BRANCH
 cli/dotnet-install.sh -i cli -c $DOTNET_BRANCH
@@ -67,8 +81,8 @@ then
 fi
 
 # restore packages
-echo "$DOTNET msbuild build/build.proj /t:Restore /p:VisualStudioVersion=16.0 /p:Configuration=Release /p:BuildNumber=1 /p:ReleaseLabel=beta"
-$DOTNET msbuild build/build.proj /t:Restore /p:VisualStudioVersion=16.0 /p:Configuration=Release /p:BuildNumber=1 /p:ReleaseLabel=beta
+echo "dotnet msbuild build/build.proj /t:Restore /p:VisualStudioVersion=16.0 /p:Configuration=Release /p:BuildNumber=1 /p:ReleaseLabel=beta"
+dotnet msbuild build/build.proj /t:Restore /p:VisualStudioVersion=16.0 /p:Configuration=Release /p:BuildNumber=1 /p:ReleaseLabel=beta
 
 if [ $? -ne 0 ]; then
 	echo "Restore failed!!"
@@ -76,8 +90,8 @@ if [ $? -ne 0 ]; then
 fi
 
 # run tests
-echo "$DOTNET msbuild build/build.proj /t:CoreUnitTests /p:VisualStudioVersion=16.0 /p:Configuration=Release /p:BuildNumber=1 /p:ReleaseLabel=beta"
-$DOTNET msbuild build/build.proj /t:CoreUnitTests /p:VisualStudioVersion=16.0 /p:Configuration=Release /p:BuildNumber=1 /p:ReleaseLabel=beta
+echo "dotnet msbuild build/build.proj /t:CoreUnitTests /p:VisualStudioVersion=16.0 /p:Configuration=Release /p:BuildNumber=1 /p:ReleaseLabel=beta"
+dotnet msbuild build/build.proj /t:CoreUnitTests /p:VisualStudioVersion=16.0 /p:Configuration=Release /p:BuildNumber=1 /p:ReleaseLabel=beta
 
 if [ $? -ne 0 ]; then
 	echo "Tests failed!!"

--- a/build/common.ps1
+++ b/build/common.ps1
@@ -155,8 +155,10 @@ Function Install-DotnetCLI {
     )
     $vsMajorVersion = Get-VSMajorVersion
     $MSBuildExe = Get-MSBuildExe $vsMajorVersion
-    $CliBranchListForTesting = & $msbuildExe $NuGetClientRoot\build\config.props /v:m /nologo /t:GetCliBranchForTesting
-    $CliBranchList = $CliBranchListForTesting.Trim().Split(';');
+
+    $CmdOutLines = ((& $msbuildExe $NuGetClientRoot\build\config.props /v:m /nologo /t:GetCliBranchForTesting) | Out-String).Trim()
+    $CliBranchListForTesting = ($CmdOutLines -split [Environment]::NewLine)[-1]
+    $CliBranchList = $CliBranchListForTesting -split ';'
 
     $DotNetInstall = Join-Path $CLIRoot 'dotnet-install.ps1'
 
@@ -171,7 +173,7 @@ Function Install-DotnetCLI {
 
     ForEach ($CliBranch in $CliBranchList) {
         $CliBranch = $CliBranch.trim()
-        $CliChannelAndVersion = $CliBranch -split "\s+"
+        $CliChannelAndVersion = $CliBranch -split ":"
 
         $Channel = $CliChannelAndVersion[0].trim()
         if ($CliChannelAndVersion.count -eq 1) {

--- a/build/config.props
+++ b/build/config.props
@@ -34,7 +34,7 @@
     <!-- Specifies the SDK version to download to use for testing. The first value represents the channel, the second value represents the exact SDK version to be download. If a version is not specified, the latest version from the channel will be downloaded. 
     Note that multiple SDKs can be downloaded by using `;` as a separator.
     -->
-    <CliBranchForTesting Condition="'$(CliBranchForTesting)' == ''">release/5.0.2xx 5.0.200-servicing.21120.4</CliBranchForTesting>
+    <CliBranchForTesting Condition="'$(CliBranchForTesting)' == ''">release/5.0.2xx:5.0.200-servicing.21120.4</CliBranchForTesting>
   </PropertyGroup>
 
   <!-- Config -->

--- a/scripts/funcTests/runFuncTests.sh
+++ b/scripts/funcTests/runFuncTests.sh
@@ -59,7 +59,8 @@ echo "dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting"
 IFS=$'\n'
 CMD_OUT_LINES=(`dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting`)
 # Take only last the line which has the version information and strip all the spaces
-DOTNET_BRANCHES=${CMD_OUT_LINES[-1]//[[:space:]]}
+CMD_LAST_LINE=${CMD_OUT_LINES[@]:(-1)}
+DOTNET_BRANCHES=${CMD_LAST_LINE//[[:space:]]}
 unset IFS
 
 IFS=$';'

--- a/scripts/funcTests/runFuncTests.sh
+++ b/scripts/funcTests/runFuncTests.sh
@@ -25,6 +25,8 @@ pushd $DIR/
 
 mono --version
 
+dotnet --info
+
 # Download the CLI install script to cli
 echo "Installing dotnet CLI"
 mkdir -p cli
@@ -36,7 +38,12 @@ chmod +x scripts/funcTests/dotnet-install.sh
 
 # Get recommended version for bootstrapping testing version
 # Issue 8936 - DISABLED TEMPORARILY cli/dotnet-install.sh -i cli -c 2.2
-scripts/funcTests/dotnet-install.sh -i cli -c 2.2 -NoPath
+scripts/funcTests/dotnet-install.sh -i cli -c 2.2 -nopath
+
+if (( $? )); then
+	echo "The .NET CLI Install failed!!"
+	exit 1
+fi
 
 DOTNET="$(pwd)/cli/dotnet"
 
@@ -60,10 +67,14 @@ do
 	echo "Version is: $Version"
 	scripts/funcTests/dotnet-install.sh -i cli -c $Channel -v $Version -nopath
 
-	# Display current version
-	$DOTNET --version
-	dotnet --info
+	if (( $? )); then
+		echo "The .NET CLI Install for $DOTNET_BRANCH failed!!"
+		exit 1
+	fi
 done
+
+# Display .NET CLI info
+$DOTNET --info
 
 echo "initial dotnet cli install finished at `date -u +"%Y-%m-%dT%H:%M:%S"`"
 
@@ -104,6 +115,7 @@ fi
 # restore packages
 echo "dotnet msbuild build/build.proj /t:Restore /p:VisualStudioVersion=16.0 /p:Configuration=Release /p:BuildNumber=1 /p:ReleaseLabel=beta"
 dotnet msbuild build/build.proj /t:Restore /p:VisualStudioVersion=16.0 /p:Configuration=Release /p:BuildNumber=1 /p:ReleaseLabel=beta
+
 if [ $? -ne 0 ]; then
 	echo "Restore failed!!"
 	exit 1

--- a/scripts/funcTests/runFuncTests.sh
+++ b/scripts/funcTests/runFuncTests.sh
@@ -47,14 +47,24 @@ fi
 
 DOTNET="$(pwd)/cli/dotnet"
 
+# Let the dotnet cli expand and decompress first if it's a first-run
+$DOTNET --info
 
+# Get CLI Branches for testing
 echo "dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting"
-# run it twice so dotnet cli can expand and decompress without affecting the result of the target
-dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting
-DOTNET_BRANCHES="$(dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting)"
-echo $DOTNET_BRANCHES | tr ";" "\n" |  while read -r DOTNET_BRANCH
+
+IFS=$'\n'
+CMD_OUT_LINES=(`dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting`)
+# Take only last the line which has the version information and strip all the spaces
+DOTNET_BRANCHES=${CMD_OUT_LINES[-1]//[[:space:]]}
+unset IFS
+
+IFS=$';'
+for DOTNET_BRANCH in ${DOTNET_BRANCHES[@]}
 do
 	echo $DOTNET_BRANCH
+
+	IFS=$':'
 	ChannelAndVersion=($DOTNET_BRANCH)
 	Channel=${ChannelAndVersion[0]}
 	if [ ${#ChannelAndVersion[@]} -eq 1 ]
@@ -63,6 +73,8 @@ do
 	else
 		Version=${ChannelAndVersion[1]}
 	fi
+	unset IFS
+
 	echo "Channel is: $Channel"
 	echo "Version is: $Version"
 	scripts/funcTests/dotnet-install.sh -i cli -c $Channel -v $Version -nopath

--- a/scripts/funcTests/runFuncTests.sh
+++ b/scripts/funcTests/runFuncTests.sh
@@ -45,6 +45,9 @@ if (( $? )); then
 	exit 1
 fi
 
+# Disable .NET CLI Install Lookup
+DOTNET_MULTILEVEL_LOOKUP=0
+
 DOTNET="$(pwd)/cli/dotnet"
 
 # Let the dotnet cli expand and decompress first if it's a first-run

--- a/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs
+++ b/test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs
@@ -39,7 +39,9 @@ namespace Dotnet.Integration.Test
             var dotnetExecutableName = RuntimeEnvironmentHelper.IsWindows ? "dotnet.exe" : "dotnet";
             TestDotnetCli = Path.Combine(_cliDirectory, dotnetExecutableName);
 
-            var sdkPath = Directory.EnumerateDirectories(Path.Combine(_cliDirectory, "sdk")).Single();
+            var sdkPath = Directory.EnumerateDirectories(Path.Combine(_cliDirectory, "sdk"))
+                            .Where(d => !string.Equals(Path.GetFileName(d), "NuGetFallbackFolder", StringComparison.OrdinalIgnoreCase))
+                            .Single();
 
             MsBuildSdksPath = Path.Combine(sdkPath, "Sdks");
 
@@ -442,6 +444,7 @@ SDKs found: {string.Join(", ", Directory.EnumerateDirectories(sdkDir).Select(Pat
             var artifactsDirectory = DotnetCliUtil.GetArtifactsDirectoryInRepo();
             var pathToSdkInCli = Path.Combine(
                     Directory.EnumerateDirectories(Path.Combine(cliDirectory, "sdk"))
+                        .Where(d => !string.Equals(Path.GetFileName(d), "NuGetFallbackFolder", StringComparison.OrdinalIgnoreCase))
                         .First());
             const string configuration =
 #if DEBUG

